astropy/io/votable/tree.py

```diff
@@ -2459,11 +2460,12 @@ def ref(self, ref):
                 ref = None
             else:
                 self._fields = table.fields
+                self._all_fields = table.all_fields
                 self._params = table.params
                 self._groups = table.groups
                 self._links = table.links
         else:
-            del self._fields[:]
+            del self.all_fields[:]
```

Comment:
Should the underscore be here?
```suggestion
            del self._all_fields[:]
```

Reply:
I also don't grok why `del` is called here for a bunch of stuff, or how all the attribute access would be affected afterwards.

Reply:
yup, missing an underscore indeed, thanks !
I spent a couple minutes here trying to understand it myself but frankly I don't think I understood it either when I was working on this. It also seems like the branch is never hit in tests so it might very well be plain broken or out of sync. I would still consider cleaning it out of scope here, as this is meant to be a minimal refactor as far as internals are concerned. What do you think ?

Reply:
Yeah that is fine. Thanks!